cleanup/cleanDockerImages/cleanDockerImages.groovy

```diff
@@ -82,37 +89,72 @@ def simpleTraverse(parentInfo, oldSet, imagesPathMap, imagesCount) {
         // - implement daysPassed policy first and delete the images that
         //   qualify
         // - aggregate the image info to group by image and sort by create
+        //   (byDownloadDate=false) or downloaded/updated (byDownloadDate=true)
         //   date for maxCount policy
-        if (checkDaysPassedForDelete(childItem)) {
+        if (checkDaysPassedForDelete(childItem, byDownloadDate)) {
             log.debug("Adding to OLD MAP: $parentRepoPath")
             oldSet << parentRepoPath
         } else if ((maxCount = getMaxCountForDelete(childItem)) > 0) {
             log.debug("Adding to IMAGES MAP: $parentRepoPath")
-            def parentCreatedDate = parentInfo.created
             def parentId = parentRepoPath.parent.id
             def oldmax = maxCount
             if (parentId in imagesCount) oldmax = imagesCount[parentId]
             imagesCount[parentId] = maxCount > oldmax ? maxCount : oldmax
             if (!imagesPathMap.containsKey(parentId)) {
                 imagesPathMap[parentId] = []
             }
-            imagesPathMap[parentId] << [parentRepoPath, childItem.created]
+            def itemLastUsedDate = getItemLastUsedDate(childItem, byDownloadDate)
+            imagesPathMap[parentId] << [parentRepoPath, itemLastUsedDate]
         }
         break
     }
 }
 
+// Check Last Downloaded Date for a specified path and return the value as
+// long (epoch).
+/// Returns 0 when item was never downloaded.
+def getLastDownloadedDate(itemPath) {
+    def lastDownloadedDate = 0
+    def itemStats = repositories.getStats(itemPath)
+    if (itemStats) {
+        lastDownloadedDate = itemStats.getLastDownloaded()
+        log.debug("lastDownloadedDate: STAT lastDownloadedDate =${lastDownloadedDate} (${new Date(lastDownloadedDate)})")
+    } else {
+        log.debug("NO STATS for ${itemPath} found")
+    }
+    return lastDownloadedDate
+}
+
+// Retrieve and return item last use date as 'long' (epoch)
+// By default this is the item creation date (original plugin behaviour).
+// When 'byDownloadDate' this will be either last download date, falling back to 'modified' date
+def getItemLastUsedDate(item, byDownloadDate=false) {
```

Comment:
Is this default to `false` needed? AFAICT this function never gets called without passing the parameter, since the `checkDaysPassedForDelete()` function which wraps it explicitly passes it in

Reply:
It's not needed. I put it in as default to emphasize that it's the default, originall behaviour but on second thought I agree - it's just redundant.

Reply:
Default removed.